src/index.js

```diff
@@ -1,52 +1,67 @@
-import {List, Map} from 'immutable';
-
 export const BEGIN = '@@optimist/BEGIN';
 export const COMMIT = '@@optimist/COMMIT';
 export const REVERT = '@@optimist/REVERT';
 
 export const ensureState = state => {
-  if (Map.isMap(state)) {
-    if (List.isList(state.get('history'))) {
-      return state.get('current');
+  if (state instanceof Object) {
+    if (Array.isArray(state.history)) {
+      return state.current;
     }
   }
   return state;
 };
 
-export const preloadState = state => Map({
+export const preloadState = state => ({
   beforeState: undefined,
-  history: List(),
+  history: [],
   current: state
 });
 
+const shift = arr => {
```

Comment:
I might move these methods to a separate module (eg. `immutable-array-utils.js`) and add some tests just for completeness.